src/source/geojson_worker_source.js

```diff
@@ -120,13 +124,19 @@ class GeoJSONWorkerSource extends VectorTileWorkerSource {
      * @param params.source The id of the source.
      * @param callback
      */
-    loadData(params: LoadGeoJSONParameters, callback: Callback<void>) {
+    loadData(params: LoadGeoJSONParameters, callback: Callback<boolean>) {
```

Comment:
Don't the `params` and `callback` need to be stored on a per-source basis as well?

I think the root problem here is that `WorkerSource`s should be per-source, not per-source-type. The mapping from source ID to `WorkerSource` instance should be handled by `Worker`.

Reply:
> Don't the params and callback need to be stored on a per-source basis as well?

Oh, of course. Good catch.

> I think the root problem here is that WorkerSources should be per-source, not per-source-type. The mapping from source ID to WorkerSource instance should be handled by Worker.

🤔 That sounds convincing, although I'd like to handle it in a different PR. On a quick inspection, it looks like the `StyleLayerIndex` is the main thing that would remain shared between multiple `WorkerSource`s?